lib/bundler/rubygems_integration.rb

```diff
@@ -39,7 +39,12 @@ def loaded_specs(name)
     end
 
     def mark_loaded(spec)
-      Gem.loaded_specs[spec.name] = spec
+      name = spec.name
+      if existing = Gem.loaded_specs[name]
+        existing.activated = false
```

Comment:
Heh, yeah. Pretty sure that `activated` was invented later than some of the RubyGems versions we support.